src/main/java/com/gregtechceu/gtceu/api/machine/trait/NotifiableEnergyContainer.java

```diff
@@ -313,6 +313,11 @@ public List<EnergyStack> handleRecipeInner(IO io, GTRecipe recipe, List<EnergySt
                 continue;
             }
 
+            // Check if recipe voltage is less than or equal to handler voltage
+            if (io.support(IO.IN) && stack.voltage() > this.getInputVoltage()) {
```

Comment:
also check the output value:
```suggestion
            long maxVoltage = io == IO.IN ? this.getInputVoltage() : this.getOutputVoltage();
            if (stack.voltage() > maxVoltage) {
```